drivers/storage/portworx/portworx.go

```diff
@@ -244,14 +244,19 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 			toUpdate.Status.DesiredImages.CSISnapshotter = release.Components.CSISnapshotter
 		}
 
-		if toUpdate.Spec.Monitoring != nil &&
-			toUpdate.Spec.Monitoring.Prometheus != nil &&
-			toUpdate.Spec.Monitoring.Prometheus.Enabled &&
-			(toUpdate.Status.DesiredImages.PrometheusOperator == "" || pxVersionChanged) {
-			toUpdate.Status.DesiredImages.Prometheus = release.Components.Prometheus
-			toUpdate.Status.DesiredImages.PrometheusOperator = release.Components.PrometheusOperator
-			toUpdate.Status.DesiredImages.PrometheusConfigMapReload = release.Components.PrometheusConfigMapReload
-			toUpdate.Status.DesiredImages.PrometheusConfigReloader = release.Components.PrometheusConfigReloader
+		if toUpdate.Spec.Monitoring != nil && toUpdate.Spec.Monitoring.Prometheus != nil {
+			if toUpdate.Spec.Monitoring.Prometheus.Enabled &&
+				(toUpdate.Status.DesiredImages.PrometheusOperator == "" || pxVersionChanged) {
+				toUpdate.Status.DesiredImages.Prometheus = release.Components.Prometheus
+				toUpdate.Status.DesiredImages.PrometheusOperator = release.Components.PrometheusOperator
+				toUpdate.Status.DesiredImages.PrometheusConfigMapReload = release.Components.PrometheusConfigMapReload
+				toUpdate.Status.DesiredImages.PrometheusConfigReloader = release.Components.PrometheusConfigReloader
+			}
+			if toUpdate.Spec.Monitoring.Prometheus.AlertManager != nil &&
+				toUpdate.Spec.Monitoring.Prometheus.AlertManager.Enabled &&
+				(toUpdate.Status.DesiredImages.AlertManager == "" || pxVersionChanged) {
```

Comment:
Do we have test case for PX version upgrade in the testrail?

Reply:
Yes, we do have a bunch of different upgrade scenarios including the version upgrade.
Upgrade cases - https://portworx.testrail.net/index.php?/suites/view/1&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=3481
Version upgrade - https://portworx.testrail.net/index.php?/cases/view/50278